json_test.go

```diff
@@ -19,6 +19,31 @@ import (
 	"testing"
 )
 
+func TestJSONIntegerRanges(t *testing.T) {
+	// This value is in range so it should be fine with both room versions
+	input := `{"foo": 9007199254740991}`
+	if _, err := EnforcedCanonicalJSON([]byte(input), RoomVersionV1); err != nil {
+		// Room version 1 allows this value
+		t.Fatalf("should be valid")
```

Comment:
Prefer `Errorf` here so it runs the other checks. Also please include the `err` in the log output.